lib/diamond/utils.ts

```diff
@@ -0,0 +1,305 @@
+import { FacetFilter, selectorIsFiltered, validateFilters } from './filters';
+import { AddressZero } from '@ethersproject/constants';
+import { Contract } from '@ethersproject/contracts';
+import {
+  IDiamondReadable,
+  IDiamondWritable,
+} from '@solidstate/typechain-types';
+
+export enum FacetCutAction {
+  ADD,
+  REPLACE,
+  REMOVE,
+}
+
+export interface Facet {
+  target: string;
+  selectors: string[];
+}
+
+export interface FacetCut extends Facet {
+  action: FacetCutAction;
+}
+
+// returns a list of signatures for a contract
+export function getSignatures(contract: Contract): string[] {
```

Comment:
Rename to `getFuntionSignatures` so as not to imply that event signatures, etc, are included.

Reply:
https://github.com/solidstate-network/solidstate-solidity/pull/158/commits/9d92ea0a7f5d5231717e8a2cd41de95fcb1f70c5